cmip6_downscaling/config/config.py

```diff
@@ -1,104 +1,197 @@
-    env_config = {
-        "AZURE_STORAGE_CONNECTION_STRING": return_azure_config()["connection_string"],
-        "EXTRA_PIP_PACKAGES": return_extra_pip_packages()["EXTRA_PIP_PACKAGES"],
-        'OMP_NUM_THREADS': '1',
-        'MPI_NUM_THREADS': '1',
-        'MKL_NUM_THREADS': '1',
-        'OPENBLAS_NUM_THREADS': '1',
-    }
-    return env_config
-
-
-def return_kubernetes_run_config():
-    kubernetes_run_config = KubernetesRun(
-        cpu_request=return_kubernetes_config()["kubernetes_cpu"],
-        memory_request=return_kubernetes_config()["kubernetes_memory"],
-        image=return_docker_config()["image"],
-        labels=return_prefect_config()["agent"],
-        env=return_env_config()["env_config"],
-    )
-    return kubernetes_run_config
-
-
-def return_dask_executor():
-
-    dask_executor = DaskExecutor(
-        cluster_class=lambda: KubeCluster(
-            make_pod_spec(
-                image=return_docker_config()["image"],
-                memory_limit=return_dask_config()["dask_executor_memory_limit"],
-                memory_request=return_dask_config()["dask_executor_memory_request"],
-                threads_per_worker=return_dask_config()["dask_executor_threads_per_worker"],
-                cpu_limit=return_dask_config()["dask_executor_cpu_limit]"],
-                cpu_request=return_dask_config()["dask_executor_cpu_request"],
-                env=return_env_config()["env_config"],
-            )
-        ),
-        adapt_kwargs={
-            "minimum": return_dask_config()["dask_executor_adapt_min"],
-            "maximum": return_dask_config()["dask_executor_adapt_max"],
-        },
-    )
-    return dask_executor
+from prefect.executors import DaskExecutor, LocalDaskExecutor, LocalExecutor  # noqa: F401
+from prefect.run_configs import KubernetesRun, LocalRun
+from prefect.storage import Azure, Local
+
+# Add new config that is hybrid or local compute, but has prefect storage access (ie. what I've been using to debug. Local is now non-write permissions.)
+
+
+class BaseConfig:
+    """Base configuration class that defines abstract methods (storage, run_config and executor) for subclasses.
+
+    Attributes
+    ----------
+
+    Methods
+    -------
+    storage()
+        abstract definition for storage method
+    run_config()
+        abstract definition for run_config method
+    executor()
+        abstract definition for executor method
+    """
+
+    @property
+    @abstractmethod
+    def storage(self) -> Any:  # pragma: no cover
+        pass
+
+    @property
+    @abstractmethod
+    def run_config(self) -> Any:  # pragma: no cover
+        pass
+
+    @property
+    @abstractmethod
+    def executor(self) -> Any:  # pragma: no cover
+        pass
+
+
+class CloudConfig(BaseConfig):
+    def __init__(self, **kwargs):
+        self.connection_string = os.environ.get("AZURE_STORAGE_CONNECTION_STRING")
+
+        self.intermediate_cache_path = "az://flow-outputs/intermediates"
```

Comment:
is the intention to hard-code these paths as defaults? i feel like pulling that out could be useful.

Reply:
Ah totally. Any thoughts on a good place for them?

Reply:
maybe as a dictionary at the top that's specific to cmip6? though this definitely isn't necessary to merge - maybe just for future reference we could change it